src/NHibernate.Test/NHSpecificTest/GH1391/Fixture.cs

```diff
@@ -0,0 +1,154 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml;
+using NHibernate.Cfg;
+using NHibernate.Impl;
+using NHibernate.Util;
+using NUnit.Framework;
+using Environment = NHibernate.Cfg.Environment;
+
+namespace NHibernate.Test.NHSpecificTest.GH1391
+{
+	public class Fixture
+	{
+		private readonly Random _random = new Random();
+		
+		[Test]
+		public void Concurrent()
+		{
+			// Simulating two session factories, where one has tracking enabled and the other disabled
+			Parallel.For(0, 100, i =>
+			{
+				if (_random.Next(2) == 0)
+				{
+					Enabled();
+				}
+				else
+				{
+					Disabled();
+				}
+			});
+		}
+
+		[Test]
+		public void ConcurrentAsync()
```

Comment:
Shouldn't the "...Async" tests be generated?

Reply:
Currently, the generator is not able to convert a [Parallel.For](https://github.com/maca88/AsyncGenerator/issues/63) and [Thread.Sleep](https://github.com/maca88/AsyncGenerator/issues/29), but I will make sure that those will be possible with the next release.